52.md

```diff
@@ -0,0 +1,142 @@
+NIP-52
+======
+
+Calendar Events
+---------------
+
+`draft` `optional` `author:tyiu`
+
+This specification defines calendar events representing an occurrence at a specific moment or between moments. These calendar events are replaceable and referenceable per [NIP-33](33.md) and deletable per [NIP-09](09.md).
+
+Unlike the term `calendar event` specific to this NIP, the term `event` is used broadly in all the NIPs to describe any Nostr event. The distinction is being made here to discern between the two terms.
+
+## Calendar Events
+
+### Format
+
+The format uses a parameterized replaceable event kind `31923` with a list of tags as described below:
+* `d` (required) universally unique identifier (UUID). Generated by the client creating the calendar event.
+* `summary` (required) name of the calendar event
+* `description` (optional) detailed description of the calendar event
+* `start` (required) inclusive start Unix timestamp in seconds. Must be less than `end`, if it exists.
+* `end` (optional) exclusive end Unix timestamp in seconds
+* `location` (optional) location of the calendar event. e.g. address, GPS coordinates, meeting room name, link to video call
+* `g` (optional) [geohash](https://en.wikipedia.org/wiki/Geohash) to associate calendar event with a searchable physical location as suggested as an example by [NIP-12](12.md)
+* `p` (optional, repeated) 32-bytes hex pubkey of a participant
+* `t` (optional, repeated) hashtag to categorize calendar event
+* `r` (optional, repeated) references / links to web pages, documents, video calls, recorded videos, etc.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <Unix timestamp in seconds>,
+  "kind": "31923",
+  "tags": [
+    ["d", "<UUID>"],
+
+    ["summary", "<summary of calendar event>"],
+    ["description", "<description of calendar event>"],
+
+    // Timestamps
+    ["start", "<Unix timestamp in seconds>"],
+    ["end", "<Unix timestamp in seconds>"],
+
+    // Location
+    ["location", "<location>"],
+    ["g", "<geohash>"],
+
+    // Participants
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>"],
+    ["p", "<32-bytes hex of a pubkey>", "<optional recommended relay URL>"],
+
+    // Hashtags
+    ["t", "<tag>"],
+    ["t", "<tag>"],
+
+    // Reference links
+    ["r", "<url>"],
+    ["r", "<url>"]
+  ]
+}
+```
+
+Calendar events are not required to be part of a [calendar](#calendar).
+
+## Calendar Event RSVP
```

Comment:
as this is spec, and since an implementation could implement calendar events and possibly not  RSVP, it would be more accurate to place the latter in another RFC. just my 21 sats.

Reply:
I swapped the kind numbers between calendars and RSVPs so that calendar events and calendars could be sequentially next to each other in the same NIP-52, and then I split off RSVPs into NIP-53.

Reply:
I reversed this decision. @fiatjaf said the two could be merged into one.